purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -4353,17 +4353,22 @@ class PurchasesTest {
         type: ProductType,
         offeringIdentifier: String?
     ): ReceiptInfo {
-        val productDetails =
-            if (type == ProductType.SUBS) createMockProductDetailsFreeTrial(productId, 2.00)
-            else createMockOneTimeProductDetails(productId, 2.00)
+        if (type == ProductType.SUBS) {
+            val productDetails = createMockProductDetailsFreeTrial(productId, 2.00)
 
-        val defaultOfferDetails = productDetails.subscriptionOfferDetails?.first { it.isBasePlan }
-        val storeProduct = if (type == ProductType.SUBS) productDetails.toStoreProduct(
-            productDetails.subscriptionOfferDetails!!,
-            defaultOfferDetails
-        ) else productDetails.toInAppStoreProduct()
+            val defaultOfferDetails = productDetails.subscriptionOfferDetails?.first { it.isBasePlan }
+            val storeProduct = productDetails.toStoreProduct(
+                productDetails.subscriptionOfferDetails!!,
+                defaultOfferDetails
+            )
 
-        return mockQueryingProductDetails(storeProduct, offeringIdentifier)
+            return mockQueryingProductDetails(storeProduct, offeringIdentifier)
+        } else {
+            val productDetails = createMockOneTimeProductDetails(productId, 2.00)
+            val storeProduct = productDetails.toInAppStoreProduct()
+
+            return mockQueryingProductDetails(storeProduct, offeringIdentifier)
+        }
```

Comment:
The inline if statements made this a little hard to follow and debug so split it up into two clear paths for `SUBS` and `INAPP`